File: www/messy/cgi-bin/wow.php
#!/usr/bin/php
<?php
// En CGI, PHP gère souvent les headers tout seul, 
// mais on peut forcer le header avec la fonction header() ou un echo.
echo "Content-Type: text/html; charset=utf-8\r\n\r\n";
?>
<!DOCTYPE html>
<html lang="fr">
<head>
    <meta charset="UTF-8">
    <style>
        body {
            background-color: #050505;
            color: #00ff41; /* Vert Matrix */
            font-family: 'Courier New', monospace;
            padding: 20px;
            line-height: 1.4;
        }
        .matrix-container {
            border: 1px solid #00ff41;
            padding: 20px;
            box-shadow: 0 0 15px #00ff41;
            background: rgba(0, 255, 65, 0.05);
        }
        h1 {
            text-align: center;
            text-transform: uppercase;
            letter-spacing: 5px;
            border-bottom: 2px solid #00ff41;
            padding-bottom: 10px;
        }
        .env-var {
            margin: 5px 0;
            display: flex;
        }
        .key { color: #fff; font-weight: bold; min-width: 250px; }
        .val { color: #00ff41; word-break: break-all; }
        .blink { animation: blinker 1s linear infinite; }
        @keyframes blinker { 50% { opacity: 0; } }
    </style>
</head>
<body>
    <div class="matrix-container">
        <h1><span class="blink">></span> PHP_CGI_EXECUTED</h1>
        
        <p><strong>Date du système :</strong> <?php echo date('Y-m-d H:i:s'); ?></p>
        <p><strong>PHP Version :</strong> <?php echo phpversion(); ?></p>
        
        <hr style="border: 0.5px solid #00ff41;">
        
        <h3>ENVIRONNEMENT CGI :</h3>
        <?php
        // On récupère toutes les variables d'environnement passées par le serveur
        $env_vars = [
            'SERVER_SOFTWARE', 'SERVER_NAME', 'GATEWAY_INTERFACE',
            'REQUEST_METHOD', 'QUERY_STRING', 'REMOTE_ADDR', 
            'HTTP_USER_AGENT', 'SCRIPT_NAME'
        ];

        foreach ($env_vars as $var) {
            $value = getenv($var) ?: 'NOT_SET';
            echo "<div class='env-var'>";
            echo "<span class='key'>$var :</span>";
            echo "<span class='val'>$value</span>";
            echo "</div>";
        }
        ?>
    </div>
</body>
</html>

File: www/messy/cgi-bin/wow.py
#!/usr/bin/env python3 -u
import os
import sys
import platform
import datetime
import subprocess

# On force le CRLF pour être ultra-compatible avec ton handler
CRLF = "\r\n"

def get_uptime():
    try:
        uptime = subprocess.check_output(['uptime', '-p']).decode().strip()
    except:
        uptime = "Non disponible"
    return uptime

# 1. Headers
print(f"Content-Type: text/html; charset=utf-8{CRLF}", end="")
print(CRLF, end="")

# 2. Le HTML/CSS
html_content = f"""
<!DOCTYPE html>
<html lang="fr">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Python CGI Dashboard</title>
    <style>
        :root {{
            --bg: #0f172a;
            --card: rgba(30, 41, 59, 0.7);
            --accent: #38bdf8;
            --text: #f1f5f9;
        }}
        body {{
            background: radial-gradient(circle at top left, #1e293b, #0f172a);
            color: var(--text);
            font-family: 'Segoe UI', system-ui, sans-serif;
            display: flex;
            justify-content: center;
            align-items: center;
            min-height: 100vh;
            margin: 0;
        }}
        .container {{
            background: var(--card);
            backdrop-filter: blur(10px);
            border: 1px solid rgba(255, 255, 255, 0.1);
            padding: 2rem;
            border-radius: 24px;
            box-shadow: 0 25px 50px -12px rgba(0, 0, 0, 0.5);
            width: 90%;
            max-width: 500px;
            text-align: center;
        }}
        h1 {{
            color: var(--accent);
            font-size: 1.5rem;
            margin-bottom: 1.5rem;
            text-transform: uppercase;
            letter-spacing: 2px;
        }}
        .stat-grid {{
            display: grid;
            gap: 1rem;
        }}
        .stat-item {{
            background: rgba(15, 23, 42, 0.5);
            padding: 1rem;
            border-radius: 12px;
            display: flex;
            justify-content: space-between;
            align-items: center;
            border-left: 4px solid var(--accent);
        }}
        .label {{ opacity: 0.7; font-size: 0.9rem; }}
        .value {{ font-weight: bold; color: #fff; }}
        .pulse {{
            width: 10px;
            height: 10px;
            background: #22c55e;
            border-radius: 50%;
            display: inline-block;
            box-shadow: 0 0 10px #22c55e;
            animation: blink 2s infinite;
        }}
        @keyframes blink {{
            0%, 100% {{ opacity: 1; }}
            50% {{ opacity: 0.3; }}
        }}
    </style>
</head>
<body>
    <div class="container">
        <div style="margin-bottom: 1rem;"><span class="pulse"></span> LIVE CGI STATUS</div>
        <h1>System Monitor</h1>
        
        <div class="stat-grid">
            <div class="stat-item">
                <span class="label">OS</span>
                <span class="value">{platform.system()} {platform.release()}</span>
            </div>
            <div class="stat-item">
                <span class="label">Node</span>
                <span class="value">{platform.node()}</span>
            </div>
            <div class="stat-item">
                <span class="label">Python Version</span>
                <span class="value">{platform.python_version()}</span>
            </div>
            <div class="stat-item">
                <span class="label">Server Time</span>
                <span class="value">{datetime.datetime.now().strftime('%H:%M:%S')}</span>
            </div>
            <div class="stat-item">
                <span class="label">Uptime</span>
                <span class="value">{get_uptime()}</span>
            </div>
        </div>

        <p style="margin-top: 2rem; font-size: 0.8rem; opacity: 0.5;">
            Généré via execve() • Protocole CGI/1.1
        </p>
    </div>
</body>
</html>
"""
print(html_content)
